### PR TITLE
feat: IndexingMath, Proxy, Migratable

### PR DESCRIPTION
**Pull request by deluca-mike**

### CODE REVIEW

File: src/ERC20Extended.sol [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// SPDX-License-Identifier: GPL-3.0
+// SPDX-License-Identifier: MIT
```
Comment (toninorair) [inline src/ERC20Extended.sol]: License needs to stay GPL for now

File: src/libs/ContractHelper.sol [hunk 2]
```diff
@@ -35,14 +35,38 @@ library ContractHelper {
...
-                                ? abi.encodePacked(bytes1(0xd6), bytes1(0x94), account_, uint8(nonce_))
-                                : nonce_ <= 0xff
-                                ? abi.encodePacked(bytes1(0xd7), bytes1(0x94), account_, bytes1(0x81), uint8(nonce_))
-                                : nonce_ <= 0xffff
-                                ? abi.encodePacked(bytes1(0xd8), bytes1(0x94), account_, bytes1(0x82), uint16(nonce_))
-                                : nonce_ <= 0xffffff
-                                ? abi.encodePacked(bytes1(0xd9), bytes1(0x94), account_, bytes1(0x83), uint24(nonce_))
-                                : abi.encodePacked(bytes1(0xda), bytes1(0x94), account_, bytes1(0x84), uint32(nonce_))
+                                    ? abi.encodePacked(bytes1(0xd6), bytes1(0x94), account_, uint8(nonce_))
+                                    : nonce_ <= 0xff
```
Comment (toninorair) [inline src/libs/ContractHelper.sol:39]: what is the reason for this change? can it be in a separate Pr with a short info

Comment (deluca-mike) [reply src/libs/ContractHelper.sol:39]: prettier auto formatting

File: src/ERC20Extended.sol [hunk 3]
```diff
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: GPL-3.0
 
-pragma solidity 0.8.26;
+pragma solidity >=0.8.20 <0.9.0;
```
Comment (toninorair) [inline src/ERC20Extended.sol:3]: 👍

**Review by PierrickGT — APPROVED** [review]

**Review by toninorair — APPROVED** [review]
